.github/workflows/ccpp.yml

```diff
@@ -46,7 +46,10 @@ jobs:
     - name: Build
       run: |
            cd gccrs-build; \
-           make -j $(nproc)
+           make -j $(nproc) > log 2>&1;
+           cat log;
+           cat log | grep 'warning:' > warnings;
+           diff warnings ../expected_warnings
```

Comment:
As the warnings may appear in random order, I suppose we really need to `sort` them, as I had mentioned.  (Then also `sort` the baseline file.)

May want to address <https://en.wikipedia.org/wiki/Cat_(Unix)#Useless_use_of_cat>.

I suggest to rename the `warnings` file to `log-warnings` or similar.

I suggest to rename the `expected_warnings` file to `bors-log-expected_warnings` or similar, so as to give more context what this file is about -- as it'll live in the GCC/Rust repository.

I suggest to swap the `diff` arguments: convention is to show old vs. new, here: expected vs. actual.

Instead of plain `diff` I suggest `diff -U0`, so that we get the standard uni-diff header printed (so that users recognize what the following output is about), but without any uni-diff context (as that's not useful here). 

As the `make` now no longer is the last command, we'll loose its exit status (relevant in case of build failure), so we'll have to tune the shell scripting to preserve that.

Reply:
@CastilloDel, please let us know if you'd like help with the shell scripting -- not everyone's favorite, I heard.  ;-)

Reply:
If using `sort` you may need to force the locale to have a stable sort across the CI and dev's hosts?

Reply:
@dkm, I was expecting this to only run with bors "in the could", which I assume to have deterministic setup/behavior.  But yeah, for local use (and also local updating of the `bors-log-expected_warnings`!), doing this with `LC_ALL=C` or similar would make a lot of sense, yes.

Reply:
> As the `make` now no longer is the last command, we'll loose its exit status (relevant in case of build failure), so we'll have to tune the shell scripting to preserve that.

Actually, not sure about that -- I don't know how these workflow files are interpreted, exactly.

What should work, is to simply move the `cat log` into its own *step*, and move the `grep` and `diff` into another *step*?

Reply:
> @dkm, I was expecting this to only run with bors "in the could", which I assume to have deterministic setup/behavior. But yeah, for local use (and also local updating of the `bors-log-expected_warnings`!), doing this with `LC_ALL=C` or similar would make a lot of sense, yes.

Yes, I was thinking about what would happen when someone needs to update the refs. A bit like what happened with clang-format behaving differently btw CI and local.